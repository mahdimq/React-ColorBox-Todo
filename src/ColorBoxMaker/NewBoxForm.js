import React, { useState } from 'react';

function NewBoxForm({ addBox }) {
	const INITIAL_STATE = {
		width: '',
		height: '',
		backgroundColor: ''
	};

	const [formData, setFormData] = useState(INITIAL_STATE);

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((formData) => ({
			...formData,
			[name]: value
		}));
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		addBox({ ...formData });
		setFormData(INITIAL_STATE);
	};

	return (
		<form onSubmit={handleSubmit}>
			<label htmlFor='width'>Width</label>
			<input
				id='width'
				type='number'
				name='width'
				min={0}
				placeholder='Box Width'
				value={formData.width}
				onChange={handleChange}
			/>
			<label htmlFor='height'>Height</label>
			<input
				id='height'
				type='number'
				name='height'
				min={0}
				placeholder='Box Height'
				onChange={handleChange}
				value={formData.height}
			/>
			<label htmlFor='backgroundColor'>Background Color</label>
			<input
				id='backgroundColor'
				type='text'
				name='backgroundColor'
				placeholder='Background Color'
				onChange={handleChange}
				value={formData.backgroundColor}
			/>
			<button>Add Box</button>
		</form>
	);
}

export default NewBoxForm;
