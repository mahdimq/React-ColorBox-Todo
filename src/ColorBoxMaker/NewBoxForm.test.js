import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';

// SMOKE TEST
it('renders component without crashing', () => {
	render(<NewBoxForm />);
});

// SNAPSHOT TEST
it('matches snapshot', () => {
	const { asFragment } = render(<NewBoxForm />);
	expect(asFragment()).toMatchSnapshot();
});
