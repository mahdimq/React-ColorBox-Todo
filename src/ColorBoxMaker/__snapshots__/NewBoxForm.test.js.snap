// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width
    </label>
    <input
      id="width"
      min="0"
      name="width"
      placeholder="Box Width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height
    </label>
    <input
      id="height"
      min="0"
      name="height"
      placeholder="Box Height"
      type="number"
      value=""
    />
    <label
      for="backgroundColor"
    >
      Background Color
    </label>
    <input
      id="backgroundColor"
      name="backgroundColor"
      placeholder="Background Color"
      type="text"
      value=""
    />
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
